src/core/types/object/json.js

```diff
@@ -16,30 +16,29 @@ let JSONScalarDef;
 
 /**
  * JSON object. It's a map with string keys and JSON values.
- * @typedef {!Object<string, ?*>} (* should be JSONValueDef)
+ * @typedef {Object<string, ?*>} (* should be JSONValueDef)
  */
 let JSONObjectDef;
 
 /**
  * JSON array. It's an array with JSON values.
- * @typedef {!Array<?*>} (* should be JSONValueDef)
+ * @typedef {Array<?*>} (* should be JSONValueDef)
  */
 let JSONArrayDef;
```

Comment:
nit: should you move all these typedef names to the docblock like you did elsewhere in the file?

Reply:
I could, and I did in some places in this PR, but it doesn't really matter/change anything

Reply:
Agreed. Just about consistency at this point (and I like the idea that the comment can't be validly used in any actual code)